File: backend/src/main/java/br/com/calciolari/datahub/imports/domain/parser/ParsedImportTotals.java
package br.com.calciolari.datahub.imports.domain.parser;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * Totals declared by the source report and/or derived strictly from parsed
 * records. Calculated fields are provenance {@code CALCULATED_DATA}.
 */
public record ParsedImportTotals(
		BigDecimal sourceQuantityTotal,
		BigDecimal parsedQuantityTotal,
		BigDecimal sourceRevenueTotal,
		BigDecimal parsedRevenueTotal,
		LocalDateTime firstMovementAt,
		LocalDateTime lastMovementAt
) {
	public static ParsedImportTotals empty() {
		return new ParsedImportTotals(null, null, null, null, null, null);
	}
}

File: backend/src/Calciolari.DataHub/Imports/Domain/Parser/ParsedImportTotals.cs
namespace Calciolari.DataHub.Imports.Domain.Parser;

/// <summary>
/// Totals declared by the source report and/or derived strictly from parsed
/// records. Calculated fields are provenance CALCULATED_DATA.
/// </summary>
public sealed record ParsedImportTotals(
    decimal? SourceQuantityTotal,
    decimal? ParsedQuantityTotal,
    decimal? SourceRevenueTotal,
    decimal? ParsedRevenueTotal,
    DateTime? FirstMovementAt,
    DateTime? LastMovementAt)
{
    public static ParsedImportTotals Empty { get; } = new(null, null, null, null, null, null);
}
